### ISSUE TITLE
FSLeyes does not open selected image when used as default application in MacOs Big Sur

### ISSUE THREAD

**Issue by JulianKlug** [pauldmccarthy/fsleyes#67]
Hello everyone,

Since updating to MacOs Big Sur, I have had the issue that when opening a nifti file with fsleyes selected as the default application, fsleyes starts but does not display the image. 

Os: MacOs Big Sur 11.1 
FSLeyes version: 0.34.2
FSL version: 6.0.4
OpenGL version: 2.1
OpenGL compatibility: 2.1
OpenGL renderer: AMD Radeon Pro 5300M OpenGL Engine

Thank you for your support and your great software!

**Comment by pauldmccarthy** [comment]
Hi @JulianKlug I think the first thing I would try would be to delete and re-install the `$FSLDIR/fslpython` environment, as something may have been broken during the OS upgrade:

1. Remove `$FSLDIR/fslpython` by running `sudo rm -r $FSLDIR/fslpython`
2. Re-install by running `sudo $FSLDIR/etc/fslconf/fslpython_install.sh -f $FSLDIR`

Step 2 will take a little while, depending on the speed of your internet connection

**Comment by JulianKlug** [comment]
Unfortunately this did not fix the issue ...

**Comment by pauldmccarthy** [comment]
Can you try starting FSLeyes from a terminal (just open a terminal and type `fsleyes`), and see if any messages are printed when you try to open an image?

**Comment by JulianKlug** [comment]
This works correctly and no log is displayed ...

**Comment by pauldmccarthy** [comment]
So to clarify:

 - when you run `fsleyes` from a terminal, you are successfully able to load and display an image
 - when you start FSLeyes via the desktop (e.g. clicking on the FSLeyes icon, or opening a file through Finder), FSLeyes opens, but the image is not displayed.

Is that correct?

It's a long shot, but there is a chance that macOS is caching some information about the old version of FSLeyes. You might be able to force a cache refresh like so:

```
cd /Applicatsions/FSLeyes.app
mv Contents/Info.plist ./Info.plist.backup
# This command will fail
./Contents/MacOS/fsleyes
mv ./Info.plist.backup Contents/Info.plist
# This command should work
./Contents/MacOS/fsleyes
```

Does that fix the image display?

**Comment by JulianKlug** [comment]
Weirdly the command does not fail even when I move the `Info.plist` 


> ```
> # This command will fail
> ./Contents/MacOS/fsleyes

**Comment by pauldmccarthy** [comment]
@JulianKlug Can you elaborate a little more on the problem? You say that "FSLeyess starts but does not display the image" - does FSLeyes *load* the image, but not display it? Or is the image not even loaded?

**Comment by JulianKlug** [comment]
So when clicking on the image to be opened with FSLeyes, FSL opens but does neither load nor display the image (see screenshot attached)

Once fsleyes is open, it works as expected, ie I can still load and display images correctly.  


![image](https://user-images.githubusercontent.com/8020367/106784957-31dcc900-664d-11eb-8f5d-548845cb42a3.png)

**Comment by pauldmccarthy** [comment]
Ok then, the problem is not as serious as I'd thought - I'd initially interpreted your description as being that the image was loaded, but you got a black display. 

I have managed to reproduce this on an old macbook that I have remote access to (haven't been able to collect it because of covid), and can confirm that it is a bug in FSLeyes. My old macbook is running Catalina, so some behavioural change must have crept in in a previous maOS release, which caused the macOS desktop/Finder integration to break.

I have a growing list of FSLeyes issues that I need to tend to, so will try and make time to fix this and release a new version in the next couple of weeks.

**Comment by JulianKlug** [comment]
Great! Thanks for your amazing work!

**Comment by DimitraF** [comment]
I have the same issue here. I also noticed that the application name on the top right corner does not say FSLeyes but python.

<img width="1293" alt="Screenshot 2021-03-02 at 16 28 04" src="https://user-images.githubusercontent.com/43215599/109663014-5062b100-7b74-11eb-862f-84b7089bbe6a.png">

**Comment by pauldmccarthy** [comment]
This issue has been fixed, and I'm hoping to get a new release out in the next couple of weeks. 

Unfortunately, I don't think I can do anything about the application name being "python", due to the way that GUI applications are executed from a conda environment on macOS. I will do a little more experimenting on thta though.

**Comment by pauldmccarthy** [comment]
Hopefully this issue should be resolved as of the latest version of FSLeyes (1.0.3). I still don't have physical access to a mac, and my remote desktop connection is unbearably slow, so mac testing is not really possible at this point in time. Please get back to me if you're still having trouble though
